### DISCUSSION TITLE
help with classpath

### DISCUSSION THREAD

**Discussion by gsiawGH** [microsoft/vscode-java-debug#1070] [Q&A] [answered]
i have a maven/java project. i want to debug the app and load a specific properties file.

when i run with a barebones launch.json file, it starts and immediately dies, because it cant find the property file. 

 seems to autogenerate a jar file to load, 
` -cp /var/folders/l_/8934fty54l1c_bn_6dtkwxy00000gp/T/cp_6ucrk3lwp5r30l8hwi7wyxntg.jar  `

however, i have no idea whats in the jar file, and i cant even locate it. does it delete this jar file when the debugger ends? and how is it even generating this path/jar? does it generate this jar from my maven pom.xml ? 

when i add a classpath into my launch.json file,  like this:

`classPaths": ["/Users/a/b/c/d/mainframe-sync/src/main/resources"],`

i see this reflected properly in the terminal

`-cp /Users/a/b/c/d/mainframe-sync/src/main/resources`

however, of course i need to add the path to the java files ,  so i add this to my launch.json
`"classPaths": ["/Users/a/b/c/d/mainframe-sync/src/main/resources", "/Users/a/b/c/d/mainframe-sync/src/main/java/"],`

but then the terminal goes back to the autogenerated classpath (?!)

`-cp /var/folders/l_/8934fty54l1c_bn_6dtkwxy00000gp/T/cp_85bi0lceunskfkyk041k2xgwr.jar `

im running VSCode Version: 1.62.0
with vscjava.vscode-java-debug v0.36.0

is this a maven issue? not sure why the debugger is behaving this way. doesnt seem to be documented  in https://code.visualstudio.com/docs/java/java-debugging

**Comment by testforstephen** [comment, marked as answer]
Since some terminals (such as cmd) have a limit on the max length of a command line, the debugger will generate the classpath args into a tmp file by default to shorten the length of the command line. That's why you see a tmp jar file cp_xyz.jar being passed as classpath. It's generate based on the classpath of your pom.xml'. You could turn off the tmp file by adding a config `"shortenCommandLine": "none"` to launch.json.

If you want to manually configure `classPaths` parameter in launch.json, you could refer to https://github.com/microsoft/vscode-java-debug/issues/93#issuecomment-848463025.

**Reply by gsiawGH** [reply]
omg, thank you so much!
